jgrapht-core/src/main/java/org/jgrapht/util/SuperSplitQueue.java

```diff
@@ -0,0 +1,397 @@
+     * otherwise it is empty
+     * 
+     * @param universeSize size of the universe
+     * @param isFull true if the new SuperSplitQueue should be filled
+     */
+    private SuperSplitQueue(int universeSize, boolean isFull)
+    {
+        this(universeSize);
+
+        if (!isFull) {
+            return;
+        }
+
+        for (int i = 0; i < universeSize; i++) {
+            addLast(i, 0);
+        }
+    }
+
+    /**
+     * Initializes a SuperSplitQueue with the given ordering
+     * 
+     * @param universeSize size of the universe
+     * @param sortedElements the ordering
+     */
+    private SuperSplitQueue(int universeSize, int[] sortedElements)
+    {
+        this(universeSize);
+        for (int i : sortedElements) {
+            addLast(i, 0);
+        }
+    }
+
+    /**
+     * adds a new empty SubSplitQueue with consecutive index.
+     * 
+     * @return new empty SubSplitQueue
+     */
+    private SubSplitQueue addNewSubSplitQueue()
+    {
+        SubSplitQueue result = new SubSplitQueue(amountQueues, this);
+
+        firstOfQ.add(-1);
+        lastOfQ.add(-1);
+        sizeOfQ.add(0);
+        queueByIndex.add(result);
+
+        amountQueues++;
+
+        return result;
+    }
+
+    /**
+     * Returns the index with the lowest order in the SubSplitQueue with index queueIndex. This
+     * element won't be removed
+     * 
+     * @param queueIndex the index of the SubSplitQueue
+     * @return first item in SubSplitQueue with index queueIndex
+     */
+    int peek(int queueIndex)
+    {
+        if (isEmpty(queueIndex)) {
+            throw new NoSuchElementException();
+        }
+        return firstOfQ.get(queueIndex);
+    }
+
+    /**
+     * Returns and deletes the index with the lowest order in the SubSplitQueue with index
+     * queueIndex.
+     * 
+     * @param queueIndex the index of the SubSplitQueue
+     * @return irst item in SubSplitQueue with index queueIndex
+     */
+    int poll(int queueIndex)
+    {
+        int result = peek(queueIndex);
+        remove(result, queueIndex);
+        return result;
+    }
+
+    /**
+     * Returns true IFF the given SubSplitQueue has no elements
+     * 
+     * @param queueIndex Index of the SubSplitQueue
+     * @return Returns whether SubSplitQueue is empty
+     */
+    boolean isEmpty(int queueIndex)
+    {
+        return firstOfQ.get(queueIndex) == -1;
+    }
+
+    /**
+     * Removes the given element and keeps the data structure consistent Runs in constant time
+     * 
+     * @param element the element to be removed
+     */
+    void remove(int element, int queueIndex)
+    {
+        inputChecker(element, queueIndex);
+        if (containedIn[element] != queueIndex) {
+            throw new NoSuchElementException("Element " + element + " not in specified queue");
+        }
+        // for h <-> element <-> j, where h = previous[element] and j = next[element]
```

Comment:
1) dont substitute previous and next. Thats more confusing
2) what is the semantic of <-> ?!